fix up enterprise licensing documentation

nslater opened the pull request:
our enterprise licensing documentation (in the source code) was not clear. this pr is an attempt to address shortcomings

changes:

- delete LICENSE and NOTICE files for `jmx-monitoring`, in favour of an `enterprise` directory level NOTICE file
- update wording of NOTICE file for clarity
- added an "all rights reserved" line to the NOTICE file
- update wording of top-level NOTICE file for overall consistency and clarity
- remove unnecessary boilerplate in build file (which erroneously identified the file as belonging to an enterprise "module")

jodok commented on NOTICE:
```diff
@@ -10,10 +10,9 @@ Different parts of CrateDB are available under different licences.
-CrateDB also includes enterprise features. These features are marked in the
-documentation and in their respective source files, and are not licensed under
-the Apache License. To use these components, you must purchase a commercial
-licence agreement with Crate.
```
is there a specific reason to remove that?

nslater replied:
I have replaced it with a rewritten para in the same location

see here: https://github.com/crate/crate/pull/7472/files#diff-aa14599dc3fdfa9cc81a7c3c247b77b9R13

basically what I've done here is standardize on the word "module" to mean "the chunks of code" (as we refer to them as "modules" elsewhere) and then clarify that the modules provide "features" (which we use in our wording of the legal text)

I have then shortened this and ended the para by saying that the modules are not available under an Apache license

I dropped the "To use these components, you must purchase a commercial license agreement with Crate." part, because the very next paragraph repeats the same thing

see https://github.com/crate/crate/blob/b498ded52555d5bb0783518959b6d351535ae8a4/NOTICE#L17

jodok replied:
got it! thanks.

nslater commented:
@jodok I removed the LICENSE file from the `enterprise` directory because in essence, the license that governs these files is *only available if you buy it from us*. so what we're doing with the `NOTICE` file here is saying "okay anything under this directory, is NOT covered by our top-level LICENSE, and you can't get a replacement license unless you buy it from us"

jodok commented:
what about about adding a LICENSE file that says "proprietary" - or some more blurb? wouldn't this be more explicit?

nslater commented:
@jodok how about https://github.com/crate/crate/pull/7472/commits/4011731ff03c8cc58ca17a7fd653ee4fb058110b

nslater commented:
@jodok okay summary of my changes:

- moved the NOTICE file to a LICENSE file under `enterprise`
- copied the exact same text from this LICENSE file to a new section in the top-level LICENSE file, which is exactly what we would do if this were third-party components we were bundling
- updated the top level NOTCE to basically point you to the top level LICENSE file for licensing information about the enterprise modules (which is again, what we would do if we were handling someone else's code)

the only thing that remains to be decided upon is this:

- if we take this to it's logical conclusion, i.e. do exactly what we would do if we were bundling some third-party modules, we would remove the LICENSE file from the `enterprise` directory. because the whole purpose of the top-level LICENSE file is to collect all the LICENSE-style notices from around the code base and put them in one place (deleting them from where they originally were)

thoughts? comments?

jodok commented on LICENSE:
```diff
@@ -202,15 +202,41 @@
 
+================================================================================
+
+For the `enteprise` directory:
```
enterprise

nslater replied:
the backticks here are for consistency with the other entry in this file

see here: https://github.com/crate/crate/blob/588259c7e4e26b81ef336fef39d42e596c104702/LICENSE#L232

using backticks to quote the directory name as a literal

jodok replied:
i meant the spelling of enterprise :)

jodok commented on enterprise/LICENSE:
```diff
@@ -0,0 +1,19 @@
+
+This directory contains proprietary enterprise modules. These modules (which in
+turn provide enterprise features) are not licensed under any version of the
+Apache License.
```
maybe also for the other mentions - not make it specific on apache only?

nslater commented:
retest this please
